### ISSUE TITLE
Youtube videos stop playing.

### ISSUE THREAD

**Issue by madhatr** [brave/browser-laptop#8590]
- Did you search for similar issues before submitting this one?
Yes, similar ones existed but were more about videos not starting.

- Describe the issue you encountered:
While playing longer youtube videos; if I am in a different tab for a while and come back to the youtube tab the video stops playing and the loading(buffering) symbol shows up even though the video is buffered for a good 3 minutes ahead.  When I attempt to click a different time in the stream Brave crashes/freezes and reloads all tabs with the youtube tab loading fine except the video stream portion is now green.  Refreshing the tab with F5 fixes the green stream. 

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 10 Home x64

- Brave Version (revision SHA):
Brave: 0.15.1 
rev: ea5024d1f3e47d34de79b70e3cd74f1408c985ce 
Muon: 2.58.8 
libchromiumcontent: 58.0.3029.81 
V8: 5.8.283.32 
Node.js: 7.9.0 
Update Channel: dev 
os.platform: win32 
os.release: 10.0.14393 
os.arch: x64

- Steps to reproduce:
    1. Play youtube video for a while, seems to be random amounts of time(longer the better chance to reproduce)
    2. Be in another tab
    3. Return to youtube video tab
    
- Actual result:
Video stops playing

- Expected result:
Tab opens and everything is okay.

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
No idea

- Can this issue be consistently reproduced?
Due to the time factor in waiting, no it can not.

- Any related issues:
The green screen makes me think something is conflicting with the video driver possibly.
This also occurs in Chrome, yet never in Edge.

**Comment by alexwykoff** [comment]
I have tried 5 and 20 minute intervals, but neither have worked for me. 
I have tried with both the video paused before switching tabs and playing continuously. 

@madhatr From how I read your initial report, it seemed as though the video was pre-recorded (not a live stream) and that it was playing continuously. Is that the case?

Also how many other tabs did you have open? Were any of them in a private state or pinned?

**Comment by madhatr** [comment]
Yes it was one of those hour plus music videos.  I'll attempt to make it happen again.  It did it twice to me in a few hours.

**Comment by luixxiul** [comment]
if you report us back, it would be helpful to let us know which video you are playing. thanks.

**Comment by madhatr** [comment]
Pretty sure it was in this playlist.   Most likely the Sith 1 hour.   Playing Stellaris you need some good conquering music.   
https://www.youtube.com/watch?v=lKYy4sT4pPM&list=PLh9R0KdDnt87qWD-KP50oB14NyK1D3a0h&index=5

**Comment by madhatr** [comment]
Okay I did a little testing and here is the info I have.

1. It only crashes when youtube is playing in another tab.  I let it run for a couple hours while gaming with the youtube tab open and it did not fail.

2. It does not fail after it has failed once and been recovered.   So once it crashes and Brave recovers and resets all tabs it will no longer fail.  It only fails after Brave has been completely closed out, and opened again.

So with info .2 does brave go into some sort of safe mode after a failure and recovery?  
Is there a log file I can inspect/upload or logging I can turn on to try and figure this out?
I will try to time it from here on out.

**Comment by Liunkae** [comment]
Crash logs are located here: `AppData\Local\Temp\Brave\Brave Developers Crashes`

**Comment by bsclifton** [comment]
With our upgrade to Chrome 58, we got a feature called tab discarding.  Tabs which Chromium determines aren't active after an amount of time will get unloaded from memory. It seems like the tab shouldn't have been discarded (since it was playing audio)... but I'm not sure how to verify that is what happened

cc: @bridiver
